File: deeds_labs/python-middleware/backend/scripts/test_ollama_direct.py
#!/usr/bin/env python3
"""Test Ollama API directly"""

import requests

models_to_test = [
    "gemma3-legal:latest",
    "gemma3-legal",
    "gemma3:270m"
]

url = "http://localhost:11434/api/generate"

for model in models_to_test:
    print(f"\n{'='*60}")
    print(f"Testing: {model}")
    print('='*60)

    payload = {"model": model, "prompt": "Hello", "stream": False}

    try:
        r = requests.post(url, json=payload, timeout=30)
        print(f"Status: {r.status_code}")

        if r.status_code == 200:
            data = r.json()
            print(f"✅ SUCCESS!")
            print(f"Response: {data.get('response', '')[:150]}")
            print(f"\n🎯 Working model: {model}")
            break
        else:
            print(f"❌ Failed: {r.text}")

    except Exception as e:
        print(f"❌ Exception: {e}")

File: scripts/tests/test-ollama-direct.mjs
/**
 * Test Ollama Direct (bypass Bifrost)
 */

async function testOllama() {
	console.log('Testing Ollama directly at http://localhost:11434...\n');

	const start = Date.now();
	const res = await fetch('http://localhost:11434/api/chat', {
		method: 'POST',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify({
			model: 'gemma4-legal:latest',
			messages: [{ role: 'user', content: 'What is negligence? Answer in 10 words.' }],
			stream: false,
		}),
	});

	const latency = Date.now() - start;
	console.log('Status:', res.status);
	console.log('Latency:', latency + 'ms');
	console.log();

	if (!res.ok) {
		const text = await res.text();
		console.error('Error:', text.slice(0, 500));
		return;
	}

	const data = await res.json();
	console.log('Response:', data.message?.content ?? 'NO CONTENT');
	console.log();
	console.log('Full structure:', JSON.stringify(data, null, 2).slice(0, 800));
}

testOllama().catch(console.error);
